### PR TITLE
Small updates to runtime.config documentation

### PR DESCRIPTION
**Pull request by swaroop-sridhar**
Update some external document references, and other minor corrections.

### CODE REVIEW

File: Documentation/specs/runtime-configuration-file.md [hunk 1]
```diff
@@ -207,7 +207,7 @@ This section is only present in the root framework's (so `Microsoft.NETCore.App`
 
 The file is read by two different components:
 
-* The host uses it to determine what to place on the TPA and Native Library Search Path lists, as well as what runtime settings to apply (GC type, etc.). See [the host spec](corehost.md).
+* The host uses it to determine what to place on the TPA and Native Library Search Path lists, as well as what runtime settings to apply (GC type, etc.). See the host [resolution](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/host-component-dependencies-resolution.md) and [probing](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/host-probing.md) documentation.
```
Comment (vitek-karas) [inline Documentation/specs/runtime-configuration-file.md]: The `host-component-dependencies-resolution.md` document is not describing the general handling of `.deps.json`. I'm afraid we don't really have a detailed spec for that - this document you're modifying is all we've got. The document you refer to here described the specifics of dependency resolution which is currently used by the `AssemblyDependencyResolver` (it used to be called `ComponentDependencyResolver`.

File: Documentation/specs/runtime-configuration-file.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 The runtime configuration files store the dependencies of an application (formerly stored in the `.deps` file). They also include runtime configuration options, such as the Garbage Collector mode. Optionally they can also include data for runtime compilation (compilation settings used to compile the original application, and reference assemblies used by the application).
 
-**Note:** This document doesn't provide full explanations as to why individual items are needed in this file. That is covered in the [host spec](corehost.md) and via the `Microsoft.Extensions.DependencyModel` assembly.
+**Note:** This document doesn't provide full explanations as to why individual items are needed in this file. That is covered in the [host documentation](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/host-component-dependencies-resolution.md) and via the `Microsoft.Extensions.DependencyModel` assembly.
```
Comment (vitek-karas) [inline Documentation/specs/runtime-configuration-file.md]: The `host-component-dependencies-resolution.md` is not general description of `.deps.json` handling... see my other comment for details.

File: Documentation/specs/runtime-configuration-file.md (hunk 1 above)
Comment (swaroop-sridhar) [reply Documentation/specs/runtime-configuration-file.md]: I agree, I tried to add the closest replacement for the non-existant reference to `corehost.md` file.
The dependency resolution/probing docs were the closest ones which process deps.json files and load assemblies. 

I can remove the reference to these files if you think that's better.

Comment (swaroop-sridhar) [reply Documentation/specs/runtime-configuration-file.md]: I changed it in this checkin: ca07a5d

**Pull request merged**